compose/internal/service/session.go

```diff
@@ -0,0 +1,86 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type (
+	session struct {
+		ctx    context.Context
+		logger *zap.Logger
+		ac     sessionAccessController
+
+		sessions map[uint64]sessionInstance
+	}
+
+	sessionInstance struct {
+		updatedAt time.Time
+		payload   interface{}
+	}
+
+	sessionAccessController interface {
+		CanReadImportSession(context.Context) bool
+		CanCreateImportSession(context.Context) bool
+		CanDeleteImportSession(context.Context) bool
+	}
+
+	SessionService interface {
+		With(ctx context.Context) SessionService
```

Comment:
Please do not follow this pattern (`With(context.Context)`) anymore. This was a design mistake. Will be refactored soon.

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

More: https://golang.org/pkg/context/

See how AccessControl, AutomationScript, AutomationTrigger are done.